### DISCUSSION TITLE
blog/how-to-selfhost-ap-at-home

### DISCUSSION THREAD

**Discussion by giscus** [pfernandez98/ap-tutorials#9] [Blog Posts Comments]
# blog/how-to-selfhost-ap-at-home

A guide to self-host Activepieces at home

https://ap-tutorials.pfernandezdev.top/blog/how-to-selfhost-ap-at-home

**Comment by MrAkber** [comment]
Hello there Pablo,
Would you be able to give us an update procedure to you great tutorial above?
I followed ActivePieces' upgrade tutorial and it went down the tube.

**Reply by MrAkber** [reply]
Just to clarify, I installed this on a RPi4 4G running on Rasp OS. Followed your Optional step.

**Reply by pfernandez98** [reply]
Hi! please share the errors you are getting so I can help you solving them. If you wanna share pictures you can use a service like `https://prnt.sc/`

**Comment by MrAkber** [comment]
No error message. I can see it on my local computer, but it is no longer accessible on the other side of CloudFlare tunnel.

**Reply by pfernandez98** [reply]
That actually happened to me the other day (no with activepieces, but with the tunnels). What I did was to uninstall the tunnel and installed it again. 

1. Go to the dashboard of the tunnels. 
1. You are gonna see a token for the installation (p. ex. `sudo cloudflared service install eyJhIjoi...`)
1. Copy that to your terminal and replace `install` for `uninstall`
1. Go back to the dashboard and click the `Refresh token` button.
1. Copy and execute the same thing you copied on step 2 (now it's gonna be another token, don't replace anything)
1. You should be good now.

**Reply by pfernandez98** [reply]
https://prnt.sc/hQTSjdOCSzLg

**Reply by MrAkber** [reply]
Thank you so very much

Is there any way, I could restore my flows?

**Reply by pfernandez98** [reply]
Your flows should have been left intact. Did you lose them?

**Reply by MrAkber** [reply]
I think I delete them with command rm .activepieces

**Reply by MrAkber** [reply]
BTW, CloudFlare is saying there is a new version available. How should I proceed?

**Comment by MrAkber** [comment]
Could you help me with a last clarification when should the AP_FRONTEND_URL stage be done?
If your answer is "post install", where do I find the .env?

**Reply by pfernandez98** [reply]
In the step #2, you should create the `.env` file. If not, check this note: https://ap-tutorials.pfernandezdev.top/blog/how-to-selfhost-ap-at-home#:~:text=Note%3A%20If%20this%20doesn%27t%20work%2C%20you%20can%20try%20renaming%20the%20.env.example%20file%20to%20.env%20and%20fill%20in%20the%20variables%20manually.

**Comment by ippo028** [comment]
hi thanks for the best tutorial around!

although i followed your tutorial in the cloudflare area, i cant seem make activepieces show itself
or it takes time to load like propagating of a website?
anyway my domain was removed from cloudflare by hostinger support since it went down suddenly they say that domain should point to nameservers of hostinger instead of cloudflare

so is it the reason that nothing shows up when i created a subdomain.mysite.com?
noob here hope you understand
thanks in advance

**Reply by ippo028** [reply]
just to add, i added a "A" record in my dns like i found in one of the youtube videos of creating a subdomain
"A" then subdomain then localhost then TTL value

i dont know where the issue is so i hope you can help me
i appreciate it thank you
